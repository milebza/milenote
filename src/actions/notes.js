import uuid from 'uuid'

// ADD_NOTE
export const addNote = ({ title = '', content = '',  date = 0 } = {}) => ({
  type: 'ADD_NOTE',
  Note: {
    id: uuid(),
    title,
    content,
    date
  }
})

// REMOVE_NOTE
export const removeNote = ({ id } = {}) => ({
  type: 'REMOVE_NOTE',
  id
})

// EDIT_NOTE
export const editNote = (id, updates) => ({
  type: 'EDIT_NOTE',
  id,
  updates
})
