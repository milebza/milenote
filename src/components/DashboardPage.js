import React from 'react'
import NoteList from './NoteList'
import NoteListFilters from './NoteListFilters'

const DashboardPage = () => (
  <div>
    <NoteListFilters />
    <NoteList />
  </div>
)

export default DashboardPage
