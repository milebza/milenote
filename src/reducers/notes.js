// Notes Reducer
const NotesReducerDefaultState = []

export default (state = NotesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_NOTE':
      return [
        ...state,
        action.note
      ]
    case 'REMOVE_NOTE':
        return state.filter(({ id }) => (id !== action.id))
    case 'EDIT_NOTE':
      return state.map((note) => {
        if (note.id === action.id) {
          return {
            ...note,
            ...action.updates
          }
        }
        return note
      })
    case 'SET_NOTES':
        return action.notes
    default:
      return state
  }
}
