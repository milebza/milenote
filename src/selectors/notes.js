import moment from 'moment'

// Get visible Notes
export default (notes, {text, startDate, endDate}) => {
  return notes.filter((note) => {
    const date = moment(note.date)
    const startDateMatch = startDate ? startDate.isSameOrBefore(date, 'day') : true
    const endDateMatch = endDate ? endDate.isSameOrAfter(date, 'day') : true
    const textMatch = note.title.toLowerCase().includes(text.toLowerCase()) || note.content.toLowerCase().includes(text.toLowerCase())

    return startDateMatch && endDateMatch && textMatch
  }).sort((a, b) => {
    // 1 -> b will go first
    return a.date < b.date ? 1 : -1
  })
}
