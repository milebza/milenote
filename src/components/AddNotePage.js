import React from 'react'
import { connect } from 'react-redux'
import { startAddNote } from '../actions/notes'
import NoteForm from './NoteForm'

export class AddNotePage extends React.Component {
  onSubmit = (note) => {
    this.props.startAddNote(note)
    this.props.history.push('/')
  }
  render() {
    return (
      <div>
        <NoteForm
          onSubmit={this.onSubmit}
        />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddNote: (note) => dispatch(startAddNote(note))
})

export default connect(undefined, mapDispatchToProps)(AddNotePage)
