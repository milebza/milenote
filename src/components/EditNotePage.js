import React from 'react'
import { connect } from 'react-redux'
import { startEditNote, startRemoveNote } from '../actions/notes'
import Note from './Note'
import NoteForm from './NoteForm'

export class EditNotePage extends React.Component {
  state = {
    editView: false
  }
  onSubmit = (note) => {
    this.props.startEditNote(this.props.note.id, note)
    this.props.history.push('/')
  }
  onRemove = () => {
    this.props.startRemoveNote({ id: this.props.note.id })
    this.props.history.push('/')
  }
  handleEditView = () => {
    this.setState((prevState) => ({ editView: true }))
  }
  render() {
    return (
      <div className="relative">
        <button className="btn-link btn--left"
          onClick={this.onRemove}><span className="icon icon-bin"></span> Remove</button>

        {
          this.state.editView ?
          (
            <NoteForm
              note={this.props.note}
              onSubmit={this.onSubmit} />
          ) :
          (
            <div>
              <button className="btn-link form__btn form__btn-edit"
                onClick={this.handleEditView}><span className="icon icon-pencil"></span> Edit</button>
              <Note {...this.props.note} />
            </div>
          )
        }
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({ note: state.notes.find((note) => note.id === props.match.params.id) })

const mapDispatchToProps = (dispatch) => ({
  startEditNote: (id, note) => dispatch(startEditNote(id, note)),
  startRemoveNote: (data) => dispatch(startRemoveNote(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditNotePage)
