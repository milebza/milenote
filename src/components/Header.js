import React from 'react'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="flex header__content-wrapper">
      <NavLink to="/" className=""><h1 className="header__title">milenote</h1></NavLink>
      <section>
        <button className="header__logout-btn" onClick={startLogout}>Logout</button>
      </section>
    </div>
  </header>
)

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)
