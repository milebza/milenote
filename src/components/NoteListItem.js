import React from 'react'
import moment from 'moment'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { startRemoveNote } from '../actions/notes'

export class NoteListItem extends React.Component {
  onRemove = () => {
    if (confirm('Trash it?')) {
      this.props.startRemoveNote({ id: this.props.id })
    }
  }
  render() {
    return (
      <div className="col col-md-4">
        <div className="note-item">
          <Link to={`/edit/${this.props.id}`}>
            <div className="note-item__content">
              <h3 className="note-item__title">{this.props.title}</h3>
              <p className="note-item__p">{this.props.content}</p>
            </div>
          </Link>
          <div className="note-item__details">
            <small className="note-item__grey">{moment(this.props.date).format("MMM Do YYYY")}</small>
            <small className="mobile-show note-item__grey note-item__dot">.</small>
            <small className="mobile-show note-item__grey" onClick={this.onRemove}>Delete</small>
            <button className="btn-link note-item__btn-remove mobile-hide" onClick={this.onRemove}><span className="icon icon-bin"></span></button>
          </div>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startRemoveNote: (data) => dispatch(startRemoveNote(data))
})

export default connect(undefined, mapDispatchToProps)(NoteListItem)
