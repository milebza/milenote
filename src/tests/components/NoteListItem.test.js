import React from 'react'
import { shallow } from 'enzyme'
import { NoteListItem } from '../../components/NoteListItem'
import notes from '../fixtures/notes'

let startRemoveNote, note, wrapper

beforeEach(() => {
  startRemoveNote = jest.fn()
  note = notes[0]
  wrapper = shallow(<NoteListItem {...note} startRemoveNote={startRemoveNote} />)
})

test('should render NoteListItem correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

// test('should handle startRemoveNote', () => {
//   wrapper.find('button').simulate('click')
//   expect(startRemoveNote).toHaveBeenLastCalledWith({id: notes[0].id})
// })
