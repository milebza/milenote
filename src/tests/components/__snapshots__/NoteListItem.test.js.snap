// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NoteListItem correctly 1`] = `
<div
  className="col col-md-4"
>
  <div
    className="note-item"
  >
    <Link
      replace={false}
      to="/edit/1"
    >
      <div
        className="note-item__content"
      >
        <h3
          className="note-item__title"
        >
          Letter
        </h3>
        <p
          className="note-item__p"
        >
          Dear friend
        </p>
      </div>
    </Link>
    <div
      className="note-item__details"
    >
      <small
        className="note-item__grey"
      >
        Jan 1st 1970
      </small>
      <small
        className="mobile-show note-item__grey note-item__dot"
      >
        .
      </small>
      <small
        className="mobile-show note-item__grey"
        onClick={[Function]}
      >
        Delete
      </small>
      <button
        className="btn-link note-item__btn-remove mobile-hide"
        onClick={[Function]}
      >
        <span
          className="icon icon-bin"
        />
      </button>
    </div>
  </div>
</div>
`;
